### PR TITLE
Use a separate job to publish the released version to SDKman

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

File: .teamcity/settings/GradleProfilerPublishing.kt [hunk 1]
```diff
@@ -34,7 +37,7 @@ object GradleProfilerPublishing : BuildType({
 
     steps {
         gradle {
-            tasks = "clean publishToSonatype closeAndReleaseSonatypeStagingRepository gitPushTag releaseToSdkMan %additional.gradle.parameters%"
+            tasks = "clean createBuildReceipt publishToSonatype closeAndReleaseSonatypeStagingRepository gitPushTag releaseToSdkMan %additional.gradle.parameters%"
```
Comment (asodja) [inline .teamcity/settings/GradleProfilerPublishing.kt:40]: Here we still have releaseToSdkMan, is that ok?

Comment (wolfs) [reply .teamcity/settings/GradleProfilerPublishing.kt]: No, it is not. I removed it.

File: buildSrc/src/main/kotlin/extensions.kt [hunk 2]
```diff
@@ -0,0 +1 @@
+const val buildReceiptName = "build-receipt.properties"
```
Comment (asodja) [inline buildSrc/src/main/kotlin/extensions.kt:1]: This value is duplicated in  buildSrc/src/main/kotlin/profiler.versioning.gradle.kts. Can it be reused?

File: buildSrc/src/main/kotlin/plugins/RootProjectVersionPlugin.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.*
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.util.Properties
+
+const val incomingBuildReceiptLocation = "incoming-distributions/$buildReceiptName"
+
+open class RootProjectVersionPlugin : Plugin<Project> {
```
Comment (asodja) [inline buildSrc/src/main/kotlin/plugins/RootProjectVersionPlugin.kt:14]: Could that be a precompiled script plugin?

File: buildSrc/src/main/kotlin/plugins/RootProjectVersionPlugin.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        target.tasks.register("createBuildReceipt") {
+            outputs.file(createdBuildReceipt).withPropertyName("buildReceipt")
+            inputs.property("version", versionInfo.version)
+            doLast {
+                createdBuildReceipt.get().asFile.writeText("version=${versionInfo.version.get()}", StandardCharsets.UTF_8)
+            }
+        }
+
+        target.gradle.taskGraph.whenReady {
+            if (hasTask(":publishToSonatype") || hasTask(":releaseToSdkMan"))
```
Comment (asodja) [inline buildSrc/src/main/kotlin/plugins/RootProjectVersionPlugin.kt:38]: `if` without `{` is easy to miss. Or I am just not used to it?

**Review by asodja — APPROVED** [review]
Added some comments, feel free to apply them if you think they make sense.

File: buildSrc/src/main/kotlin/extensions.kt (hunk 2 above)
Comment (wolfs) [reply buildSrc/src/main/kotlin/extensions.kt:1]: Well spotted! I removed it in `profiler.versioning.gradle.kts`.

File: buildSrc/src/main/kotlin/plugins/RootProjectVersionPlugin.kt (hunk 3 above)
Comment (wolfs) [reply buildSrc/src/main/kotlin/plugins/RootProjectVersionPlugin.kt]: Yeah. I converted it.

File: buildSrc/src/main/kotlin/plugins/RootProjectVersionPlugin.kt (hunk 4 above)
Comment (wolfs) [reply buildSrc/src/main/kotlin/plugins/RootProjectVersionPlugin.kt]: Good point. I added the `{`.

**Review by asodja — APPROVED** [review]

**Pull request merged**